packages/frontend/src/user/documents.tsx

```diff
@@ -0,0 +1,157 @@
+                        </table>
+                    </div>
+                </div>
+                <div class="ref-table-scroll">
+                    <table class="ref-table">
+                        <tbody>
+                            <Switch
+                                fallback={
+                                    <tr>
+                                        <td colspan="5">Loading...</td>
+                                    </tr>
+                                }
+                            >
+                                <Match when={resultOk(refStubs())}>
+                                    {(okRes) => (
+                                        <For each={okRes()}>
+                                            {(stub) => <RefStubRow stub={stub} />}
+                                        </For>
+                                    )}
+                                </Match>
+                                <Match when={resultErr(refStubs())}>
+                                    {(errRes) => (
+                                        <tr>
+                                            <td colspan="5">
+                                                Error loading documents: {errRes().message}
+                                            </td>
+                                        </tr>
+                                    )}
+                                </Match>
+                            </Switch>
+                        </tbody>
+                    </table>
+                </div>
+            </div>
+        </>
+    );
+}
+
+export function RefStubRow(props: { stub: RefStub }) {
+    const firebaseApp = useFirebaseApp();
+    const auth = getAuth(firebaseApp);
+
+    const owner = props.stub.owner;
+    const hasOwner = owner !== null;
+    const isOwner = hasOwner && auth.currentUser?.uid === owner?.id;
+    // biome-ignore lint/style/noNonNullAssertion: type narrowing doesn't work for ternary statements
+    const ownerName = hasOwner ? (isOwner ? "me" : owner!.username) : "public";
+
+    const handleClick = () => {
+        const url = getUrlForRefStub(props.stub);
+        window.location.href = url;
+    };
+
+    return (
+        <tr class="ref-stub-row" onClick={handleClick}>
+            <td>{props.stub.type_name}</td>
+            <td>{props.stub.name}</td>
+            <td>{ownerName}</td>
+            <td>{props.stub.permission_level}</td>
+            <td>
+                {new Date(props.stub.created_at).toLocaleDateString("en-US", {
+                    year: "numeric",
+                    month: "short",
+                    day: "numeric",
+                })}
+            </td>
+        </tr>
+    );
+}
+
+function getUrlForRefStub(refStub: RefStub): string {
+    const hostname = window.location.hostname;
+    const protocol = window.location.protocol;
+    const port = window.location.port;
+    return `${protocol}//${hostname}:${port}/${refStub.type_name}/${refStub.ref_id}`;
```

Comment:
I'm surprised we have to do this, namely construct the full URL here and then run `window.location.href = url` above. Can we use [`useNavigate`](https://docs.solidjs.com/solid-router/reference/primitives/use-navigate), something like:

```typescript
const navigate = useNavigate();

[...]

navigate(`/${refStub.type_name}/${refStub.ref_id}`);
```

Reply:
Yup, I changed it to this. I initially was using a `<a href=...>` element and it grew out poorly.